drivers/can/can_stm32h7.c

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2022 Blue Clover
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/can.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/clock_control.h>
+#include <drivers/pinctrl.h>
+#include <kernel.h>
+#include <stm32_ll_rcc.h>
+#include "can_mcan.h"
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(can_driver, CONFIG_CAN_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32h7_fdcan
+
+struct can_stm32h7_config {
+	struct can_mcan_msg_sram *msg_sram;
+	void (*config_irq)(void);
+	struct can_mcan_config mcan_cfg;
+	const struct pinctrl_dev_config *pcfg;
+	struct stm32_pclken pclken;
+};
+
+struct can_stm32h7_data {
+	struct can_mcan_data mcan_data;
+};
+
+static int can_stm32h7_get_core_clock(const struct device *dev, uint32_t *rate)
+{
+	ARG_UNUSED(dev);
+	int rate_tmp;
```

Comment:
```suggestion
	const uint32_t rate_tmp = LL_RCC_GetFDCANClockFreq(LL_RCC_FDCAN_CLKSOURCE);
```

Reply:
```C
/**
  * @brief  Return FDCAN clock frequency
  * @param  FDCANxSource This parameter can be one of the following values:
  *         @arg @ref LL_RCC_FDCAN_CLKSOURCE
  * @retval FDCAN clock frequency (in Hz)
  *         - @ref  LL_RCC_PERIPH_FREQUENCY_NO indicates that oscillator is not ready
  */
uint32_t LL_RCC_GetFDCANClockFreq(uint32_t FDCANxSource)
```

Reply:
I know where this is coming from :-) Fixed in tree.
#41849

Reply:
@JeremyBCD Now that #41849 is merged, could you follow-up with a similar change? I think this is the last unaddressed issue.

Reply:
Thanks @alexanderwachter, fixed.